src/Disks/IO/ReadBufferFromAzureBlobStorage.cpp

```diff
@@ -113,7 +114,9 @@ bool ReadBufferFromAzureBlobStorage::nextImpl()
     {
         try
         {
+            ResourceGuard rlock(ResourceGuard::Metrics::getIORead(), read_settings.io_scheduling.read_resource_link, to_read_bytes);
             bytes_read = data_stream->ReadToCount(reinterpret_cast<uint8_t *>(data_ptr), to_read_bytes);
+            rlock.unlock(bytes_read); // Do not hold resource under bandwidth throttler
```

Comment:
I do not understand that. 
I thought that if we use guards we set them for some wide scope and do not bother with one line code.

Reply:
I mixed it up with IOScope. `ResourceGuard` does scheduling itself.